Switch CI to GitHub actions

line-o opened the pull request:
Appveyour configuration is outdated and can potentially be replaced by adding an OS matrix to the test action.

NPM_TOKEN was added to the repository secrets

line-o committed:
chore(ci): remove travis configuration

line-o committed:
fix(ci): add github actions

- add test action that will run on pull requests
- add release action that will run on pushes to master
- add semantic-release configuration file
- add @semantic-release/github plugin
- update semantic-release to latest version

line-o committed:
fix(ci): test useful releases in appveyor

duncdrum commented:
yes ditching appveyor here would certainly help the org. Win tests for this repo on GHActions should be reasonably performant

line-o commented:
I just re-read #74 
OK, I will attempt to do it right away

line-o committed:
fix(ci): also test windows builds with GHA

- remove appveyor configuration
- remove appveyor specific scripts
- add OS matrix to github actions workflow "test"

line-o committed:
fix(ci): test workflow structure
